test/server/http/admin_test.cc

```diff
@@ -206,30 +216,68 @@ TEST_P(AdminInstanceTest, CustomHandler) {
   EXPECT_EQ(Http::Code::Accepted, getCallback("/foo/bar", header_map, response));
 }
 
-TEST_P(AdminInstanceTest, RejectHandlerWithXss) {
-  auto callback = [](absl::string_view, Http::HeaderMap&, Buffer::Instance&) -> Http::Code {
-    return Http::Code::Accepted;
+TEST_P(AdminInstanceTest, AdminStream) {
+  // Cover endStreamOnComplete().
+  // Add an handler that sets end_stream_on_complete on the AdminStream object.
+  auto callback = [](absl::string_view, Http::HeaderMap&, Buffer::Instance&,
+                     AdminStream& admin_stream) -> Http::Code {
+    admin_stream.setEndStreamOnComplete(false);
+
+    // check that value is set to false
+    if (admin_stream.endStreamOnComplete())
+      return Http::Code::NotAcceptable;
+    return Http::Code::OK;
   };
+
+  // Add non removable handler.
```

Comment:
it would be nicer to make separate testcases for these.  You could generate actually make the callback be a static method of AdminInstanceTest to share it between the methods.

Reply:
I thought it is better to cover all these setters/getters in one test.

Reply:
Common practice is to make each individual test shorter.